[Terra Form Single-Select] Announce Selected Instead of Checked (#3811)
cerner/terra-core: packages/terra-core-docs/src/terra-dev-site/doc/form-select/SingleSelect.03.doc.mdx

The Single Select component allows selecting a single option from a dropdown of selectable options.

> Consumers are encouraged to prefer "Single Select Field" component over "Single Select" component (where ever possible) as the latter does not provide labels necessary to support accessibility.

Reviewer: ```suggestion
Consumers are encouraged to use the `SingleSelectField` component over the `SingleSelect` component wherever possible, as the latter does not provide the labels necessary to support accessibility.
```

> To create an accessible Single Select component, follow the guidelines below:

Reviewer: ```suggestion
To create an accessible `SingleSelect` component:
```

> 1. Ensure that select Field can receive keyboard focus.
> 2. Ensure that Select Field is implemented correctly to programmatically associate all related content (visible label, helper text, error messaging) to the field.
> 3. Ensure the visible select Label matches any artificially applied programmatic label.

Reviewer: ```suggestion
1. Ensure that `SelectField` can receive keyboard focus.
2. Ensure that `SelectField` programmatically associates all related content (such as visible label, helper text & error messaging) to the field.
3. Ensure any visible `SelectLabel` matches any artificially applied programmatic label.
```

> 4. Error messaging should provide error suggestions when the system can programmatically understand what is wrong.

Reviewer: Are you trying to say that if a `SelectField` has an error, then suggestions to correct the error should be provided? Or that error messages should be clear and descriptive?

Author: This was added from [this code consideration](https://github.com/cerner/terra-core/blob/Form_Select/packages/terra-core-docs/src/terra-dev-site/doc/form-select/AccessibilityGuide.18.doc.mdx?plain=1#L61) under AccessibilityGuide, and referring [this](https://github.com/cerner/terra-core/blob/Form_Select/packages/terra-core-docs/src/terra-dev-site/doc/form-select/AccessibilityGuide.18.doc.mdx?plain=1#LL152C6-L152C209) it means: "Content creators must provide engineers with appropriate error messaging to account for a known invalid entry. If possible, the error messaging must provide the user guidance on how to correct the issue."
